src/Servers/Kestrel/Transport.Sockets/src/Internal/SocketConnection.cs

```diff
@@ -27,7 +27,7 @@ internal sealed class SocketConnection : TransportConnection
         private volatile bool _socketDisposed;
         private volatile Exception? _shutdownReason;
         private Task? _processingTask;
-        private readonly TaskCompletionSource _waitForConnectionClosedTcs = new TaskCompletionSource(TaskCreationOptions.RunContinuationsAsynchronously);
+        private readonly TaskCompletionSource _waitForConnectionClosedTcs = new TaskCompletionSource();
```

Comment:
I'm not sure we should do this. ConnectionContext.DisposeAsync() ultimately awaits this and that called be called from app code.

Reply:
My bad: https://github.com/dotnet/aspnetcore/blob/e7fbc332e58b969bd44db80713b8ca99299380b8/src/Servers/Kestrel/Transport.Sockets/src/Internal/SocketConnection.cs#L325